chain/chunks/src/lib.rs

```diff
@@ -775,7 +779,10 @@ impl ShardsManager {
 
     /// Returns true if transaction is not in the pool before call
     pub fn insert_transaction(&mut self, shard_id: ShardId, tx: SignedTransaction) -> bool {
-        self.tx_pools.entry(shard_id).or_insert_with(TransactionPool::new).insert_transaction(tx)
+        self.tx_pools
+            .entry(shard_id)
+            .or_insert_with(|| TransactionPool::new(self.rng_seed))
```

Comment:
Is this actually correct? It seems that all `TransactioPool`s will share the same `.rng_seed`, which seems like a change in behavior? Naively, I'd probably expect most things to store an instance of `rng`, rather than a seed.